### PR TITLE
Add optional tag name overrides to conditional tags in schema.cfg

### PR DESCRIPTION
**Pull request by rfsc-mori**
Fixes schema.cfg validation errors after #8136.

I've followed the same pattern of the other tags in this schema to allow the conditional tags `case`, `else`, `then`, `elseif` and `else` to be defined without a name key.

### CODE REVIEW

**Comment by CelticMinstrel** [comment]
Hmm. Though this seems good, we could perhaps document it a bit better by defining a "blank" type that accepts only an empty string as a valid value. That way, putting `name=foo` in those tags would still correctly be an error.

**Comment by rfsc-mori** [comment]
Done! Thanks! I've decided to use an impossible pattern so even an empty "" string won't be matched. I've also removed the comments, but I can add them back if needed. Or maybe keep them only on the first occurrence?

**Comment by rfsc-mori** [comment]
Hmm, it seems that some of these tags indeed have a name key. I'll look more into it later, sorry.

File: data/schema/schema.cfg [hunk 1]
```diff
@@ -56,6 +56,10 @@
 		name="bool"
 		value="yes|no|true|false"
 	[/type]
+    [type]
```
Comment (CelticMinstrel) [inline data/schema/schema.cfg:59]: If you think it's a good idea, we could put the explanation of "can't override a super key with its absence" here? Just a thought, I don't strongly mind one way or the other.

**Comment by rfsc-mori** [comment]
I tried to work a bit more on this and re-added some comments for documentation.

Also, please correct me if I'm wrong, I'm still learning about this part of the code (and of course in general too):
The `[elseif]` tag being a filter should be more of an `[if]` tag than a `[tag]` itself. I mean, in a sense that it should not specify `super="wml_schema/tag"` as it should not use the `[tag]` keys and tags, as this should go into its `[then]`.

Following the `[if]` pattern without the super, we can solve this by removing both the "disallowed" and the super, and then the intent feels clearer than overriding the name key with an optional value.

I feel this makes sense, but I'm not too sure if there are special cases for the `[elseif]` in this regard.

Also, if this is correct and accepted, do you think it's still a good idea to go ahead with the `[filter]` tag implementation?

File: data/schema/schema.cfg (hunk 1 above)
Comment (rfsc-mori) [reply data/schema/schema.cfg:62]: It is a great idea! Thanks! Added!

**Comment by CelticMinstrel** [comment]
> Also, please correct me if I'm wrong, I'm still learning about this part of the code (and of course in general too): The `[elseif]` tag being a filter should be more of an `[if]` tag than a `[tag]` itself. I mean, in a sense that it should not specify `super="wml_schema/tag"` as it should not use the `[tag]` keys and tags, as this should go into its `[then]`.
> 
> Following the `[if]` pattern without the super, we can solve this by removing both the "disallowed" and the super, and then the intent feels clearer than overriding the name key with an optional value.

Ah, you're right. I'd forgotten that `[elseif]` takes a `[then]` tag. So, it shouldn't have `[tag]` as its super at all.

**Comment by stevecotton** [comment]
Should this be merged for 1.18, regardless of whether #8136 is merged for 1.18?

**Comment by CelticMinstrel** [comment]
It's a pure schema change, so probably.

**Pull request merged**